ts/Extensions/Breadcrumbs.ts

```diff
@@ -0,0 +1,898 @@
+            drilldownLevels[drilldownLevels.length - 1].levelNumber || 0;
+        }
+
+        if (this.isTreemap && e) {
+            if (!breadcrumbsList[0]) {
+                // As a first element add the series.
+                breadcrumbsList.push([null, chart.series[0]]);
+            }
+            if (e.trigger === 'click') {
+                // When a user clicks add element one by one.
+                if (currentLevelNumber === null) {
+                    breadcrumbsList.push([0, chart.get(e.newRootId)]);
+                    currentLevelNumber = 0;
+                } else {
+                    breadcrumbsList.push([currentLevelNumber + 1, chart.get(e.newRootId)]);
+                }
+            } else {
+                let node = e.target.nodeMap[e.newRootId];
+                const extraNodes = [];
+
+                // When the root node is set and has parent,
+                // recreate the path from the node tree.
+                while (node.parent || node.parent === '') {
+                    extraNodes.push(node);
+                    node = e.target.nodeMap[node.parent];
+                }
+                extraNodes.reverse().forEach(function (node): void {
+                    if (currentLevelNumber === null) {
+                        breadcrumbsList.push([0, node]);
+                        currentLevelNumber = 0;
+                    } else {
+                        breadcrumbsList.push([++currentLevelNumber, node]);
+                    }
+                });
+            }
+        }
+        if (!this.isTreemap && drilldownLevels && drilldownLevels.length) {
+            // Add the initial series as the first element.
+            if (!breadcrumbsList[0]) {
+                breadcrumbsList.push([null, drilldownLevels[0].seriesOptions]);
+                currentLevelNumber = -1;
+            }
+
+            drilldownLevels.forEach(function (level): void {
+                // If level is already added to breadcrumbs list,
+                // don't add it again- drilling categories
+                if (level.levelNumber > currentLevelNumber) {
+                    breadcrumbsList.push([level.levelNumber, level.pointOptions]);
+                }
+            });
+        }
+        this.breadcrumbsList = breadcrumbsList;
+    }
+
+    /**
+     * Destroy the chosen buttons and separators.
+     *
+     * @requires  modules/breadcrumbs
+     *
+     * @function Highcharts.Breadcrumbs#destroy
+     * @param {Highcharts.Breadcrumbs} this
+     *        Breadcrumbs class.
+     * @param {Highcharts.Breadcrumbs} lastVisibleLevel
+     *        Number of levels to destoy.
+     */
+    public destroy(this: Breadcrumbs, lastVisibleLevel: number|undefined): void {
+        const breadcrumbs = this,
+            chart = breadcrumbs.chart,
+            breadcrumbsList = breadcrumbs.breadcrumbsList,
+            breadcrumbsOptions = chart.options.breadcrumbs;
```

Comment:
General issue (see also other methods):
Instead of referring to `chart.options.breadcrumbs` use `breadcrumbs.options`